### PR TITLE
Refactor FXIOS-8725 Update Fonts related to SyncNow to use FXFontStyles

### PR DESCRIPTION
**Pull request by AhmedFitoh**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8725)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/19337)

## :bulb: Description
Refactor to use `FXFontStyles` instead of `DefaultDynamicFontHelper`	

 ## Screenshots
>Before

>After

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

**Comment by cwzilla** [comment]
Looks great! Just one comment, for **No connection**, I think it can follow the same sizing as the **Syncing** text.

**Comment by AhmedFitoh** [comment]
"Syncing" is out of the scope of the ticket as it is using `LegacyDynamicFontHelper` and the scope is to change usage of  `DefaultDynamicFontHelper` to `FXFontStyles`. I took a quick look at `LegacyDynamicFontHelper` and I didn't find a match to "defaultStandardFontSize" in `FXFontStyles`, should we add it ?
@cyndichin

**Comment by cyndichin** [comment]
> "Syncing" is out of the scope of the ticket as it is using `LegacyDynamicFontHelper` and the scope is to change usage of `DefaultDynamicFontHelper` to `FXFontStyles`. I took a quick look at `LegacyDynamicFontHelper` and I didn't find a match to "defaultStandardFontSize" in `FXFontStyles`, should we add it ? @cyndichin

Thanks @AhmedFitoh! Yes, the `LegacyDynamicFontHelper` is out of scope for this ticket. I'm believe that @cwzilla meant to increase "No internet connection" to "Sync Now" font. We can leave "Syncing..." as-is for now. We can update those styles to match, I'll leave a comment to address this.

File: firefox-ios/Client/Frontend/Settings/Main/Account/SyncNowSetting.swift [hunk 1]
```diff
@@ -49,11 +49,7 @@ class SyncNowSetting: WithAccountSetting {
                 string: .FxANoInternetConnection,
                 attributes: [
                     NSAttributedString.Key.foregroundColor: theme.colors.textWarning,
-                    NSAttributedString.Key.font: DefaultDynamicFontHelper.preferredFont(
-                        withTextStyle: .body,
-                        size: 13,
-                        weight: .regular
-                    )
+                    NSAttributedString.Key.font: FXFontStyles.Regular.footnote.scaledFont()
```
Comment (cyndichin) [inline firefox-ios/Client/Frontend/Settings/Main/Account/SyncNowSetting.swift:52]: let's update this to use the body style. also if you can update the screenshot in the PR with the change, that would be great! thank you :)

**Comment by AhmedFitoh** [comment]
Now "No internet connection" is matching "Syncing", also Screenshots have been updated.
@cwzilla , cc: @cyndichin

**Review by cyndichin — APPROVED** [review]
thanks @AhmedFitoh for the contribution! Looks good on my side!

**Comment by cwzilla** [comment]
looks great, thanks @AhmedFitoh !

**Pull request merged**